### PR TITLE
Add integration tests for bulk user import logging improvements.

### PR DESCRIPTION
**Pull request by menakaj**
Depends on: wso2/carbon-identity-framework/pull/1504

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/pom.xml [hunk 1]
```diff
@@ -527,6 +527,11 @@
             <groupId>org.wso2.carbon.identity.framework</groupId>
             <artifactId>org.wso2.carbon.user.mgt.stub</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.user.mgt</artifactId>
+            <version>${carbon.identity.framework.version}</version>
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/pom.xml]: please add this dependency to parent pom with the version if not already there.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    @Test (description = "Perform bulk user import and check the bulkuserimport log file is created.")
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    public void testBulkUserImportLog() throws RemoteException, UserAdminUserAdminException {
+
+        String userStoreDomain = "PRIMARY";
+        File bulkUserFile = new File(getISResourceLocation() + File.separator + "userMgt"
+                + File.separator + "bulkUserImport.csv");
+
+        DataHandler handler = new DataHandler(new FileDataSource(bulkUserFile));
+        userMgtClient.bulkImportUsers(userStoreDomain, "bulkUserImport.csv", handler, "PassWord1@");
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: shall we define constants for "bulkUserImport.csv" and "PassWord1@" ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        super.init(TestUserMode.SUPER_TENANT_ADMIN);
+        userMgtClient = new UserManagementClient(backendURL, getSessionCookie());
+        PropertyConfigurator.configure(createLog4jConfiguration());
+    }
+
+    @Test (description = "Perform bulk user import and check the bulkuserimport log file is created.")
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    public void testBulkUserImportLog() throws RemoteException, UserAdminUserAdminException {
+
+        String userStoreDomain = "PRIMARY";
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: shall we define a constant for this?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                + File.separator + "bulkUserImport.csv");
+
+        DataHandler handler = new DataHandler(new FileDataSource(bulkUserFile));
+        userMgtClient.bulkImportUsers(userStoreDomain, "bulkUserImport.csv", handler, "PassWord1@");
+        String logsHome = Utils.getResidentCarbonHome() + File.separatorChar + "repository" + File.separatorChar +
+                "logs";
+        File logDir = new File(logsHome);
+        String[] logFiles = logDir.list();
+
+        if (logFiles != null) {
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: better to use ArrayUtils here

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            for (String fileName : logFiles) {
+                if (LOG_FILE_NAME.equals(fileName)) {
+                    isFilePresent = true;
+                }
+            }
+        }
+        Assert.assertTrue(isFilePresent);
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: better to use static assert imports

File: modules/integration/tests-integration/tests-backend/pom.xml (hunk 1 above)
Comment (menakaj) [reply modules/integration/tests-integration/tests-backend/pom.xml]: The version is taken from the parent pom.

Comment (pulasthi7) [reply modules/integration/tests-integration/tests-backend/pom.xml]: @menakaj if the version is there in the parent pom (in dependency management), it doesn't need to be included here. Refer the other dependencies.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+/**
+ * Test cases for the bulkUserImport log appender.
+ * This will check the logs are directed to the new file.
+ */
+public class UserImportLoggingTestCase extends ISIntegrationTest {
+
+    private static final Log log = LogFactory.getLog(UserImportLoggingTestCase.class);
+    private UserManagementClient userMgtClient;
+    private boolean isFilePresent = false;
+    private final String BULK_USER_FILE = getISResourceLocation() + File.separator + "userMgt"
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: let's order these such that constants will be defined first and then the class variables

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+        DataHandler handler = new DataHandler(new FileDataSource(bulkUserFile));
+        userMgtClient.bulkImportUsers(USER_STORE_DOMAIN, BULK_USER_IMPORT_ERROR_FILE, handler, null);
+
+        File logDir = new File(LOG_FILE_LOCATION);
+        String[] logFiles = logDir.list();
+        if (ArrayUtils.isNotEmpty(logFiles)) {
+            for (String fileName : logFiles) {
+                if (LOG_FILE_NAME.equals(fileName)) {
+                    isFilePresent = true;
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: shouldn't we assert something here?

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 8]
```diff
@@ -230,4 +230,9 @@
             <!--<class name="org.wso2.identity.integration.test.entitlement.EntitlementJSONSupportMultiDecisionProfileTestCase"/>-->
         </classes>
     </test>
+    <test name="is-tests-usr-mgt" preserve-order="true" parallel="false">
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Seems "is-tests-usr-mgt" is already used, Integration tests are failing with the following log,

Running TestSuite
org.apache.maven.surefire.util.SurefireReflectionException: java.lang.reflect.InvocationTargetException; nested exception is java.lang.reflect.InvocationTargetException: null
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: org.testng.TestNGException:
Two tests in the same suite cannot have the same name: is-tests-usr-mgt
	at org.testng.TestNG.checkTestNames(TestNG.java:958)
	at org.testng.TestNG.sanityCheck(TestNG.java:947)
	at org.testng.TestNG.run(TestNG.java:979)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:178)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:92)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:96)
	... 9 more

Results :

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java (hunk 7 above)
Comment (menakaj) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: Here we check the exception. 
`expectedExceptions = {AxisFault.class}`

File: modules/integration/tests-integration/tests-backend/pom.xml (hunk 1 above)
Comment (menakaj) [reply modules/integration/tests-integration/tests-backend/pom.xml]: I checked this. If we do not add the dependency version, it fails with a ClassNotFound error.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java (hunk 6 above)
Comment (menakaj) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/UserImportLoggingTestCase.java]: Done

**Comment by pulasthi7** [comment]
@menakaj , The PR is causing test failures, Please investigate and fix. Refer #3099 for details.

**Pull request merged**